### ISSUE TITLE
oh-chart-title: expression for text not evaluated / value missing

### ISSUE THREAD

**Issue by soenkekueper** [openhab/openhab-webui#1974]
## The problem

i've created a widget, that shows a pie chart. As title (in center of the pie) i want to display a value (in my case the total energy conumption). This is specified within the oh-chart-title with config param text. The parameter is working, but when i specify any kind of expression, the expression is not evaluated, insted it is printed in center of the chart.

<img width="369" alt="chart_expression" src="https://github.com/openhab/openhab-webui/assets/24605357/dbc8f6ad-54ce-4571-a61a-77616b825c81">

I've copied the widget into a local running openHAB 4 instance - there the title is missing in total.

<img width="365" alt="chart_expression_oh4" src="https://github.com/openhab/openhab-webui/assets/24605357/aed070e5-edcc-47fd-a982-c8c76f74830b">

I've also tried enclosing the expression with ' or ", both didn't change the result.

This is the widget source:

```yaml
uid: chartTestWidget
tags:
  - card
props:
  parameters:
    - default: DefaultValue
      label: Chart Title
      name: chartTitle
      required: true
      type: TEXT
timestamp: Jul 9, 2023, 3:18:39 PM
component: f7-card
config:
  title: Chart Test
slots:
  default:
    - component: f7-card-content
      slots:
        default:
          - component: oh-chart
            slots:
              title:
                - component: oh-chart-title
                  config:
                    text: =props.headerText
                    left: center
                    top: center
              series:
                - component: oh-data-series
                  config:
                    data:
                      - itemStyle:
                          color: rgb(245, 151, 69)
                        name: Some value name
                        value: 10
                      - itemStyle:
                          color: rgb(122, 198, 2459)
                        name: Another value
                        value: 30
                      - itemStyle:
                          color: rgb(64, 194, 70)
                        name: Last value
                        value: 30
                    itemStyle:
                      borderColor: "#fff"
                      borderRadius: 10
                      borderWidth: 2
                    radius:
                      - 60%
                      - 75%
                    startAngle: 45
                    type: pie
```

This is a little bit simpler than my one, i've used the expression

```
text: =(Number.parseFloat(items[props.directEnergyConsumption].state) + Number.parseFloat(items[props.batteryDischargeEnergy].state) + Number.parseFloat(items[props.importEnergy].state)).toFixed(1)'
but i think the cause is the same.
```

## Expected behavior

The specified expression should be evaluted and the result should be printed in the center of the chart.

## Steps to reproduce

1. Create a new widget
2. Copy the source code from above
3. You'll see in preview the wrong title

## Your environment

```yaml
runtimeInfo:
  version: 3.4.4
  buildString: Release Build
locale: de-DE
systemInfo:
  configFolder: /etc/openhab
  userdataFolder: /var/lib/openhab
  logFolder: /var/log/openhab
  javaVersion: 11.0.18
  javaVendor: Raspbian
  osName: Linux
  osVersion: 5.10.103-v7l+
  osArchitecture: arm
  availableProcessors: 4
  freeMemory: 34215512
  totalMemory: 204472320
  startLevel: 100
bindings:
  - astro
  - avmfritz
  - denonmarantz
  - deutschebahn
  - ecovacs
  - exec
  - heos
  - homematic
  - icalendar
  - ipp
  - modbus
  - mqtt
  - netatmo
  - network
  - ntp
  - solarforecast
  - tankerkoenig
  - tradfri
  - worxlandroid
  - xmppclient
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: false
    macos: false
    windows: true
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    os: windows
    pixelRatio: 1.5
    prefersColorScheme: light
  isSecureContext: true
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: 8
    hardwareConcurrency: 8
    language: de-DE
    languages:
      - de-DE
      - de
      - en-US
      - en
    onLine: true
    platform: Win32
  screen:
    width: 2560
    height: 1440
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: light
    filled: true
    pageTransitionAnimation: default
    bars: filled
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
  userAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36
timestamp: 2023-07-19T08:35:18.490Z
```

```yaml
runtimeInfo:
  version: 4.0.0
  buildString: "Build #3485"
locale: de-DE
systemInfo:
  configFolder: D:\openhab\conf
  userdataFolder: D:\openhab\userdata
  logFolder: D:\openhab\userdata\logs
  javaVersion: 17.0.7
  javaVendor: Eclipse Adoptium
  javaVendorVersion: Temurin-17.0.7+7
  osName: Windows 10
  osVersion: "10.0"
  osArchitecture: amd64
  availableProcessors: 8
  freeMemory: 39342240
  totalMemory: 140509184
  startLevel: 100
bindings: null
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: false
    macos: false
    windows: true
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    os: windows
    pixelRatio: 1.5
    prefersColorScheme: light
  isSecureContext: true
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: 8
    hardwareConcurrency: 8
    language: de-DE
    languages:
      - de-DE
      - de
      - en-US
      - en
    onLine: true
    platform: Win32
  screen:
    width: 2560
    height: 1440
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: light
    filled: true
    pageTransitionAnimation: default
    bars: filled
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
  userAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36
timestamp: 2023-07-19T08:36:11.172Z
```

## Browser console

none

## Browser network traffic

non failed requests

**Comment by florian-h05** [comment]
First of all, values filled into properties of the widget are not evaluated (AFAIK).

Second: I checked myself on openHAB 4.0.0, expressions work for the `title` config param of `oh-chart-title`.
The fact that nothing is displayed is normal as long as you don't configure a property named `headerText`, because `=props.headerText` evaluates to `undefined` in this case. For `undefined`, nothing is displayed.

I don't see a bug here, if you disagree, please let me know.

**Comment by soenkekueper** [comment]
Hey,
thanks for checking this.

In OH 3 this didn't work, so i created the basic example above - and that has a typo in the parameter name, so it didn't work in OH 4 too.

I've now updated to OH4 and it's working as expected. 

One last thought on this:
The documentation in https://www.openhab.org/docs/ui/building-pages.html states:

> Virtually everywhere - with the notable exception of chart pages - every time you need a config prop to be dynamically updated, you can use an expression to configure it.

As something seems to have changed, is this exception correct any more?
I didn't find a suitable issue on that...

Thanks again.

**Comment by ghys** [comment]
> >  Virtually everywhere - with the notable exception of chart pages - every time you need a config prop to be dynamically updated, you can use an expression to configure it.

> As something seems to have changed, is this exception correct any more?
I didn't find a suitable issue on that...

It's not, expressions in all chart widgets properties were implemented in #1649 and this mention in the docs wasn't updated.

**Comment by florian-h05** [comment]
@ghys Am I right that expressions are not supported in the defined properties of a self-created widget?
